### PR TITLE
More path-related cleanup (work-in-progress — not yet ready to merge, but want CI testing)

### PR DESCRIPTION
**Pull request by abertelrud**
More path-related cleanup, including adding a `PathShims.swift` file as temporary glue code until we've evolved the FileSystem API enough to switch to it.  The goal of this file is to go away once that transition is complete.

### CODE REVIEW

**Comment by abertelrud** [comment]
@swift-ci please test

File: Sources/SourceControl/GitRepository.swift [hunk 1]
```diff
@@ -15,8 +15,8 @@ import func POSIX.getenv
 import enum POSIX.Error
 import class Foundation.ProcessInfo
 
-enum GitRepositoryProviderError: Swift.Error {
-    case gitCloneFailure(url: String, path: String)
+enum GitRepositoryProviderError: ErrorProtocol {
```
Comment (ddunbar) [inline Sources/SourceControl/GitRepository.swift]: This won't build with TOT Swift, it needs to be `Swift.Error`.

**Comment by ddunbar** [comment]
Yay, looks like a great cleanup!

File: Sources/SourceControl/GitRepository.swift (hunk 1 above)
Comment (abertelrud) [inline Sources/SourceControl/GitRepository.swift]: Ah right, I missed converting that one back again.  Did I mention this PR is a work in progress...?  :)
Thanks for the heads-up.

**Comment by abertelrud** [comment]
Thanks!  Just to explain to everyone else what I'm trying to do here:  the eventual goal is to have all the code be in terms of the FileSystem protocol, so that it's agnostic about whether it's operating on the local file system, on a Git repo (perhaps even a bare one), a CPIO archive, or whatever.  But we want to evolve that API carefully and discuss it as we go.

So in the meantime, the idea is to clean up all the call sites that use a variety of file system helper functions etc and just make them all look like their familiar POSIX equivalents but be expressed in terms of AbsolutePath right now, and then to transition to FileSystem when it's available.  Those transitions should be fairly easy at that time, since the switch from String to AbsolutePath is where the disambiguation has to happen (what I've been doing now).  The switch from these shim functions to FileSystem protocol should be fairly mechanical, once it happens.

**Comment by abertelrud** [comment]
This particular PM will be updated with more cleanup diffs, with most file system access going through the shims (though I will not be modifying the code that already uses localFileSystem).  The Path class that remains in Utility will also be getting removed once all the clients are off of it.  It might be hard to see the big picture now but it should all look quite uniform once the transition to AbsolutePath is wrapped up, including removal of the old APIs.

**Comment by abertelrud** [comment]
@swift-ci please test

**Comment by abertelrud** [comment]
@swift-ci please test linux

**Comment by abertelrud** [comment]
Failures in Swift SILGen, this time.

**Pull request merged**